modules/data.atmosphere/R/download.GLDAS.R

```diff
@@ -0,0 +1,156 @@
+##' @name download.GLDAS
+##' @title download.GLDAS
+##' @export
+##' @param outfolder
+##' @param start_date
+##' @param end_date
+##' @param site_id
+##' @param lat
+##' @param lon
+##'
+##' @author Christy Rollinson
+
+download.GLDAS <- function(outfolder, start_date, end_date, site_id, lat.in, lon.in, overwrite=FALSE, verbose=FALSE, ...){  
+  require(PEcAn.utils)
+  require(RCurl)
+  require(XML)
+  require(lubridate)
+  require(ncdf4)
+  require(stringr)
+  
+  # Date stuff
+  start_date <- as.POSIXlt(start_date, tz = "GMT")
+  end_date <- as.POSIXlt(end_date, tz = "GMT")
+  start_year <- year(start_date)
+  end_year   <- year(end_date)
+  site_id = as.numeric(site_id)
+  outfolder = paste0(outfolder,"_site_",paste0(site_id %/% 1000000000, "-", site_id %% 1000000000))
+  
+  lat.in = as.numeric(lat.in)
+  lon.in = as.numeric(lon.in)
+  dap_base="http://hydro1.sci.gsfc.nasa.gov/thredds/dodsC/GLDAS_NOAH025SUBP_3H"
+  dir.create(outfolder, showWarnings=FALSE, recursive=TRUE)
+  
+  ylist <- seq(start_year,end_year,by=1)
+  rows = length(ylist)
+  results <- data.frame(file=character(rows), host=character(rows),
+                        mimetype=character(rows), formatname=character(rows),
+                        startdate=character(rows), enddate=character(rows),
+                        dbfile.name = "NLDAS",
+                        stringsAsFactors = FALSE
+                        )
+  var = data.frame(DAP.name = c("Near_surface_air_temperature","Surface_incident_longwave_radiation","Surface_pressure","Surface_incident_shortwave_radiation","Near_surface_wind_magnitude","Near_surface_specific_humidity","Rainfall_rate"),
+                   CF.name = c("air_temperature","surface_downwelling_longwave_flux_in_air","air_pressure","surface_downwelling_shortwave_flux_in_air","wind","specific_humidity","precipitation_flux"),
+                   units = c('Kelvin',"W/m2","Pascal","W/m2","m/s","g/g","kg/m2/s") 
+                   )
+  
+  for (i in 1:rows){
+    year = ylist[i]    
+    
+    # figure out how many days we're working with
+    if(rows>1 & !i==1 & !i==rows){ # If we have multiple years and we're not in the first or last year, we're taking a whole year
+      nday  = ifelse(year%%4 == 0,366,365) # leap year or not; days per year
+      days.use = 1:nday
+    } else if(rows==1){
+      # if we're working with only 1 year, lets only pull what we need to
+      nday  = ifelse(year%%4 == 0,366,365) # leap year or not; days per year
+      day1 <- yday(start_date)
+      # Now we need to check whether we're ending on the right day
+      day2 <- yday(end_date)
+      days.use = day1:day2
+      nday=length(days.use) # Update nday
+    } else if(i==1) {
+      # If this is the first of many years, we only need to worry about the start date
+      nday  = ifelse(year%%4 == 0,366,365) # leap year or not; days per year
+      day1 <- yday(start_date)
+      days.use = day1:nday
+      nday=length(days.use) # Update nday
+    } else if(i==1) {
```

Comment:
I probably won't have enough time before I go for code review, sorry! just one thing that caught my eye, aren't these conditions check the same thing? `i==1` 

also for checking leap years [I think we use](https://github.com/PecanProject/pecan/issues/801) `lubridate::leap_year` in pecan now